src/encoded/tests/test_types_file.py

```diff
@@ -376,3 +376,108 @@ def test_file_embedded_annotation_properties(testapp, file_ccre,
     res = testapp.get(file_ccre['@id'] + '@@index-data')
     assert res.json['object']['annotation_subtype'] == 'CTCF-only'
     assert res.json['object']['biochemical_inputs'] == ['cDHS', 'rDHS']
+
+
+def test_file_show_azure_uri():
+    from encoded.types.file import show_azure_uri
+    for status in File.public_s3_statuses:
+        assert show_azure_uri(
+            's3://encode-public/2020/09/26/71e9637f-bbcb-4e4b-9d92-d3acdda4bb7b/ENCFF841DAP.bigWig',
+            status,
+        )
+        assert not show_azure_uri(
+            None,
+            status,
+        )
+    for status in File.private_s3_statuses:
+        assert not show_azure_uri(
+            's3://encode-public/2020/09/26/71e9637f-bbcb-4e4b-9d92-d3acdda4bb7b/ENCFF841DAP.bigWig',
+            status,
+        )
+        assert not show_azure_uri(
+            None,
+            status,
+        )
+
+
+def test_get_key_from_s3_uri():
+    from encoded.types.file import get_key_from_s3_uri
+    s3_uri = 's3://encode-public/2020/09/26/71e9637f-bbcb-4e4b-9d92-d3acdda4bb7b/ENCFF841DAP.bigWig'
+    assert (
+        get_key_from_s3_uri(s3_uri) == '2020/09/26/71e9637f-bbcb-4e4b-9d92-d3acdda4bb7b/ENCFF841DAP.bigWig'
+    )
+
+
+def test_convert_s3_uri_to_azure_uri():
+    from encoded.types.file import convert_s3_uri_to_azure_uri
+    s3_uri = 's3://encode-public/2020/09/26/71e9637f-bbcb-4e4b-9d92-d3acdda4bb7b/ENCFF841DAP.bigWig'
+    azure_uri = convert_s3_uri_to_azure_uri(s3_uri)
+    assert azure_uri == (
+        'https://datasetencode.blob.core.windows.net/dataset/'
```

Comment:
these links don't seem to actually work....

Reply:
Sounds like bucket is synced first week of every month so only works on files released before last sync:
https://azure.microsoft.com/en-us/services/open-datasets/catalog/genomics-encode/

I thought about adding some condition to show URI based on date released being less than current month, and we still can, but wasn't sure it was worth the additional indexing time. Maybe it's too confusing to show links that are only eventually consistent though.

Reply:
I guess it would help if files had `date_released` property.